### PR TITLE
tests: source version testing: duplicate and keep old mysql- and pg-cdc tests with old syntax

### PR DESCRIPTION
**Pull request by nrainer-materialize**
This keeps original CDC tests after https://github.com/MaterializeInc/materialize/pull/29416 until the old source syntax is no longer supported.

### TODOs
Before merging, apply
* https://github.com/MaterializeInc/materialize/pull/29438 ✅ 
* https://github.com/MaterializeInc/materialize/pull/29601 ✅

### CODE REVIEW

File: ci/test/pipeline.template.yml [hunk 1]
```diff
@@ -457,6 +517,17 @@ steps:
...
+        label: Postgres RTR tests (before source versioning)
+        depends_on: build-aarch64
+        timeout_in_minutes: 30
+        inputs: [test/pg-rtr-old-syntax]
+        plugins:
+          - ./ci/plugins/mzcompose:
+              composition: pg-rtr-old-syntax
+        agents:
+          queue: hetzner-aarch64-4cpu-8gb
+
```
Comment (def-) [inline ci/test/pipeline.template.yml:530]: Can we move all the old syntax tests to nightly? I'm guessing they have lower risk to change and the new syntax tests cover a lot of the same surface area in the code.

**Review by def- — CHANGES_REQUESTED** [review]
I'm wondering if there would have been a way to achieve this without having to copy lots of test files (but now it's probably too late)

File: ci/test/pipeline.template.yml (hunk 1 above)
Comment (nrainer-materialize) [reply ci/test/pipeline.template.yml:530]: Yeah, makes sense

**Comment by nrainer-materialize** [comment]
> I'm wondering if there would have been a way to achieve this without having to copy lots of test files (but now it's probably too late)

I am open for ideas; this PR was the least time-consuming one 😀

**Comment by def-** [comment]
I haven't followed closely enough to know how closely related the semantics are, and also how long the old syntax will stick around. If the semantics are similar and the old syntax will stick around for a while, then what I'd suggest is rewriting all tests for the new syntax and adding a `--old-source-syntax` testdrive flag to have testdrive parse and rewrite the `CREATE SOURCE` queries into old syntax.

**Comment by rjobanp** [comment]
I am not sure on the best way to avoid a lot of test code duplication, but fwiw I do think there is likely to be another ~2 months while both new and old syntaxes are possible to use since we will have a fairly long migration window for customers to switch over to the new syntax. For safety reasons we should ideally keep tests for both formats during this period

**Comment by def-** [comment]
2 months I'm not that worried about, in that case I'm ok with the current approach. 2 years would be another story.

**Comment by nrainer-materialize** [comment]
Yeah, it's only supposed to be around for a limited amount of time.

File: ci/test/pipeline.template.yml (hunk 1 above)
Comment (nrainer-materialize) [reply ci/test/pipeline.template.yml]: ✅

**Review by def- — APPROVED** [review]

**Pull request merged**